lib/model/query/entities.js

```diff
@@ -171,20 +172,25 @@ createVersion.audit = (updatedEntity, dataset, partial, subDef) => (log) => {
       entityDefId: updatedEntity.aux.currentVersion.id,
       entity: { uuid: updatedEntity.uuid, dataset: dataset.name }
     });
+  return Promise.resolve();
 };
 createVersion.audit.withResult = true;
 
 ////////////////////////////////////////////////////////////////////////////////
 // WRAPPER FUNCTIONS FOR CREATING AND UPDATING ENTITIES
 
-const _createEntity = (dataset, entityData, submissionId, submissionDef, submissionDefId, event, parentEvent) => async ({ Audits, Entities }) => {
+const _createEntity = (dataset, entityData, submissionId, submissionDef, submissionDefId, event, parentEvent, forceOutOfOrderProcessing = false) => async ({ Audits, Entities }) => {
   // If dataset requires approval on submission to create an entity and this event is not
   // an approval event, then don't create an entity
   if ((dataset.approvalRequired && event.details.reviewState !== 'approved') ||
       (!dataset.approvalRequired && event.action === 'submission.update')) // don't process submission if approval is not required and submission metadata is updated
     return null;
 
-  // TODO: auto-generate a label if forced and if the submission doesn't provide one
+  // Auto-generate a label if forced and if the submission doesn't provide one
+  if (forceOutOfOrderProcessing && entityData.system.label == null) {
+    // eslint-disable-next-line no-param-reassign
+    entityData.system.label = 'auto generated';
+  }
```

Comment:
Can label be '' here instead of null? If the `<label>` tag in the submission is empty?